app/Services/ScBookingService.cs

```diff
@@ -599,20 +616,74 @@ string userDisplayName
 
             ScSessionBookingInfo bookingInfo = _session.ScBookingInfo;
 
-            // @TODO: check if timeslot is available? else return error
+            model.IsBooked = false;
+            bookingInfo.IsBooked = false;
+
+            // check if timeslot is available
+            if (bookingInfo.BookingFormula == ScFormulaType.RegularBooking)
+            {
+                // Available dates
+                List<DateTime> availableTrialDates = await GetAvailableTrialDatesAsync(
+                    ScFormulaType.RegularBooking
+                );
+
+                // check if selected date exists in the available dates
+                DateTime selectedDate = DateTime.ParseExact(
+                    bookingInfo.SelectedRegularTrialDate,
+                    "yyyy-MM-dd",
+                    System.Globalization.CultureInfo.InvariantCulture
+                );
+
+                bool dateAvailable = availableTrialDates.Contains(selectedDate);
 
-            // @TODO: book hearing? trial?
+                // thow an exception if the date is no longer available
+                if (!dateAvailable)
+                {
+                    // @TODO: throw some exception?
+                    return model;
+                }
 
-            // @TODO: store user info in session for next booking
+                // @TODO: book trial in API
+                var formula = await GetFormulaLocationAsync(
+                    bookingInfo.BookingFormula,
+                    bookingInfo.HearingBookingRegistryId,
```

Comment:
I think you actually need TrialLocation here instead of HearingBookingRegistryId.  I had it wrong in my code too initially.

Reply:
Sounds good, I updated that